addons/api/addon/serializers/host-catalog.js

```diff
@@ -98,7 +119,7 @@ export default class HostCatalogSerializer extends ApplicationSerializer {
     // Delete unnecessary fields for static host-catalog
     delete serialized.attributes;
     delete serialized.secrets;
-    delete serialized.worker_filter;
+
```

Comment:
question for understanding: why was `delete serialized.worker_filter` removed ?

Reply:
Since I added type `static` or `plugin` in the model attr via `for`, We don't need to explicitly delete this field from static serializer anymore. This case is handled in the [application serializer](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/addon/serializers/application.js#L76) where we automatically discard fields that don't belong to the record type